### PR TITLE
[Serve] unpin the starlette/fastapi version

### LINKED ISSUE #42370: [Serve] proxy failed to start as `starlette` removed `options` attribute from `starlette.middleware.Middleware`
### What happened + What you expected to happen

Proxy failed to start with error
```
File "ray/serve/_private/proxy.py", line 1202, in __init__
self.wrapped_http_proxy, **middleware.options
AttributeError: 'Middleware' object has no attribute 'options'
```

Starelette has removed this attribute after version `0.34.0`
https://github.com/encode/starlette/pull/2381/files#diff-861f6098c0edc67d2e328a0c76c86acf1cd1f4d86fc398a6838d18d69c2bdeaeL7

`pip install "starlette<=0.34.0"` has fixed it for me. `requirements` file may be updated accordingly https://github.com/ray-project/ray/blob/master/python/requirements.txt#L56

### Versions / Dependencies

Ray version 2.9.0

### Reproduction script

Run any serve app

### Issue Severity

None

### PR DESCRIPTION
**Pull request by sihanwang41**
## Related issue number

Closes: https://github.com/ray-project/ray/issues/42370

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by sihanwang41** [comment]
Hi @can-anyscale @aslonnie, PTAL.
One question: should i manually run `bazel run //release:requirements_byod_3.9.update` to update release/ray_release/byod/requirements_byod_3.9.txt by myself?

File: python/requirements_compiled.txt [hunk 1]
```diff
@@ -471,12 +471,6 @@ fairscale==0.4.6
     # via -r /ray/ci/../python/requirements/ml/tune-test-requirements.txt
 farama-notifications==0.0.4
     # via gymnasium
-fastapi==0.104.0
-    # via
-    #   -r /ray/ci/../python/requirements.txt
-    #   -r /ray/ci/../python/requirements/test-requirements.txt
-    #   aim
-    #   gradio
```
Comment (aslonnie) [inline python/requirements_compiled.txt:479]: are the packages entirely removed from the dependencies?

**Review by aslonnie — COMMENTED** [review]
the PR title says "unpin"; the PR is essentially removing the packages from deps all together?

File: python/ray/serve/_private/proxy.py [hunk 2]
```diff
@@ -1196,7 +1196,9 @@ def __init__(
 
         for middleware in http_middlewares:
             self.wrapped_http_proxy = middleware.cls(
-                self.wrapped_http_proxy, **middleware.options
+                self.wrapped_http_proxy,
+                *middleware.args,
+                **middleware.kwargs,
```
Comment (GeneDer) [inline python/ray/serve/_private/proxy.py:1201]: Thanks Sihan! But this is not backward compatible right? I think we should pin it to >= 0.35.0 or use different args depending on the `starlette` version

File: python/requirements_compiled.txt (hunk 1 above)
Comment (sihanwang41) [reply python/requirements_compiled.txt:479]: IIUC, this is the constraints file, if remove here, we will use latest version of fastapi. https://github.com/ray-project/ray/blob/master/python/setup.py#L269

Comment (aslonnie) [reply python/requirements_compiled.txt:479]: talk to @can-anyscale (or me) if you need help on how to generate / recompile this file.

**Review by aslonnie — COMMENTED** [review]
please pin the version of the dependencies in the constraint file.

File: python/requirements_compiled.txt (hunk 1 above)
Comment (can-anyscale) [reply python/requirements_compiled.txt:479]: you can update the content using this file https://buildkite.com/ray-project/premerge/builds/16643#018d149d-9eb8-4edf-a1a7-29c2e46a25dd

**Comment by can-anyscale** [comment]
do this still need review?

**Comment by edoakes** [comment]
Oh sorry I took this over and it's fixed now

**Pull request closed without merging**